Source/Core/InputCommon/ControllerEmu/ControlGroup/IMUGyroscope.cpp

```diff
@@ -162,6 +156,14 @@ std::optional<IMUGyroscope::StateData> IMUGyroscope::GetState(bool update)
   return state;
 }
 
+std::optional<IMUGyroscope::StateData> IMUGyroscope::Reset()
+{
+  // Set calibration to zero.
+  m_calibration = {};
+  RestartCalibration();
+  return std::nullopt;
```

Comment:
Bonus tip: If doing so in the ternary you've changed the code to is absolutely necessary, you could still pull it off with the comma operator. I don't the ternary operator is mission-critical in this case, though.

Reply:
I've never used the comma operator before. Could you show how you want me to fix this.

Reply:
Against my better judgement, I will show you.
```cpp
  return AreInputsBound() ? GetStateInternal() : (Reset(), std::nullopt);
```
However, I don't recommend writing code like that. The following is what I would write instead:
```cpp
  if (AreInputsBound())
    return GetStateInternal();
  Reset();
  return std::nullopt;
```
It may seem attractive to use the one-liner with the fancy language feature, but it's less readable when it uses such obscurities.

Reply:
> It may seem attractive to use the one-liner with the fancy language feature

It is the most attractive thing since magnets, but since the goal is readability, I'll go with the verbose option. :)